hilti/include/rt/types/tuple.h

```diff
@@ -17,6 +17,7 @@ template<typename Tuple, size_t Idx>
 ptrdiff_t elementOffset() {
     // This is pretty certainly not well-defined, but seems to work for us ...
     const Tuple* p = nullptr;
+    // NOLINTNEXTLINE(clang-analyzer-core.NonNullParamChecker)
```

Comment:
Yikes, I certainly agree with clang-tidy that this assured nullptr dereference is nasty, if not certainly UB. WDYT about instead of suppressing this warning to pass in the a presumably valid tuple as arg and then work with that?
```.cpp
template<typename Tuple, size_t Idx>
ptrdiff_t elementOffset(const Tuple& xs) {
    // This is pretty certainly not well-defined, but seems to work for us ...
    return reinterpret_cast<const char*>(&std::get<Idx>(xs)) - reinterpret_cast<const char*>(&xs);
}
```

Looking at the call sites we should always have such a tuple at hand.

Of course even better would be to not require passing this in, but instead work only with types, but I cannot see how that would be possible in general (it might e.g., be impossible to create an instance of the tuple since its fields might not have default constructors).

Reply:
I changed this.